### PR TITLE
Allow leaving id_number and id_type fields empty in id_info for JT6.

### PR DESCRIPTION
**Pull request by nii-mants3**
This PR introduces changes to allow `country` and `id_type` fields to be empty in `id_info` for JT6

### CODE REVIEW

File: lib/smile-identity-core/web_api.rb [hunk 1]
```diff
@@ -93,6 +93,7 @@ def id_info=(id_info)
 
       if updated_id_info[:entered] && updated_id_info[:entered] == 'true'
         %i[country id_type id_number].each do |key|
+          next if @partner_params[:job_type].to_i == JobType::DOCUMENT_VERIFICATION
```
Comment (mldangelo) [inline lib/smile-identity-core/web_api.rb]: Do we want to do this for all fields or just id_type? Why not do this above the loop rather than inside of it?

Comment (nii-mants3) [reply lib/smile-identity-core/web_api.rb]: Yeah makes sense 👍

File: lib/smile-identity-core/web_api.rb [hunk 2]
```diff
@@ -91,7 +91,7 @@ def id_info=(id_info)
       # if it's a boolean
       updated_id_info[:entered] = id_info[:entered].to_s if !updated_id_info[:entered].nil? == updated_id_info[:entered]
 
-      if updated_id_info[:entered] && updated_id_info[:entered] == 'true'
+      if updated_id_info[:entered] && updated_id_info[:entered] == 'true' && @partner_params[:job_type].to_i != JobType::DOCUMENT_VERIFICATION
```
Comment (mldangelo) [inline lib/smile-identity-core/web_api.rb]: You should check for equality instead. `!= JobType::DOCUMENT_VERIFICATION` will be more fragile than checking against known job types. `[country id_type id_number]` are required for `JobType::BIOMETRIC_KYC`, `JobType::DOCUMENT_VERIFICATION` needs `[country]`

Comment (nii-mants3) [reply lib/smile-identity-core/web_api.rb]: I'm not sure i get the equality part. I'm not checking the type here. `JobType::DOCUMENT_VERIFICATION` resolves to `6`. See https://github.com/smileidentity/smile-identity-core-ruby/blob/main/lib/smile-identity-core/constants/job_type.rb#L17

File: lib/smile-identity-core/web_api.rb [hunk 3]
```diff
@@ -91,8 +91,10 @@ def id_info=(id_info)
       # if it's a boolean
       updated_id_info[:entered] = id_info[:entered].to_s if !updated_id_info[:entered].nil? == updated_id_info[:entered]
 
-      if updated_id_info[:entered] && updated_id_info[:entered] == 'true'
-        %i[country id_type id_number].each do |key|
+      if updated_id_info[:entered] == 'true'
+        keys = %i[country id_type id_number]
+        keys -= %i[id_type id_number] if @partner_params[:job_type].to_i == JobType::DOCUMENT_VERIFICATION
```
Comment (mldangelo) [inline lib/smile-identity-core/web_api.rb]: why not 
```suggestion
  keys = if @partner_params[:job_type].to_i == JobType::DOCUMENT_VERIFICATION
    %i[country]
  else
    %i[country id_type id_number]
  end
```

File: lib/smile-identity-core/web_api.rb [hunk 4]
```diff
@@ -91,8 +91,13 @@ def id_info=(id_info)
       # if it's a boolean
       updated_id_info[:entered] = id_info[:entered].to_s if !updated_id_info[:entered].nil? == updated_id_info[:entered]
 
-      if updated_id_info[:entered] && updated_id_info[:entered] == 'true'
-        %i[country id_type id_number].each do |key|
+      if updated_id_info[:entered] == 'true'
```
Comment (mldangelo) [inline lib/smile-identity-core/web_api.rb]: I don't think this condition `(entered == 'true')` is necessary for job type 6, but I'll defer to @beaglebets. If it's not necessary, consider moving the  JobType::DOCUMENT_VERIFICATION check above entered.

**Review by mldangelo — COMMENTED** [review]
This looks good to me. Going to defer to @beaglebets for final approval. Thank you @nii-mants3

**Review by beaglebets — COMMENTED** [review]
There are some odd things happening here. `entered=true` is only applicable to biometric kyc. I think you should heed Michael's advice and do things specific to the job types they are applicable to.

**Review by mldangelo — APPROVED** [review]
thank you

**Pull request merged**